lib/asyn-thread.c

```diff
@@ -276,6 +301,9 @@ static unsigned int CURL_STDCALL getaddrinfo_thread(void *arg)
   struct thread_data *td = tsd->td;
   char service[12];
   int rc;
+#ifdef HAVE_SOCKETPAIR
+  int buf[1] = {1};
```

Comment:
This buffer declaration can be moved into the `HAVE_SOCKETPAIR` check where the code is, instead of doing it separately like this.

Also, did you make it an int and not a char on purpose? You write four bytes, not one...